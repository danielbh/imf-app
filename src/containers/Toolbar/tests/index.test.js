/**
 * Created by danielhollcraft on 9/8/17.
 */
import React from 'react';
import { shallow } from 'enzyme';
import { create } from 'react-test-renderer'
import { ButtonToolbar, ButtonGroup, Button } from 'react-bootstrap';
import { Toolbar } from '../index'
import 'jest-styled-components'

const renderComponent = (props) => shallow(
  <Toolbar {...props} />
);

describe('<Toolbar />', () => {
  it('has expected structure and styles', () => {
    const tree = create(<Toolbar />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  describe('Date range change buttons', () => {
    it('changes date range to today on click', () => {
      const selectWeek = jest.fn();
      const component = renderComponent({ selectWeek });
      component.find(Button).at(0).simulate('click');
      expect(selectWeek).toBeCalled();
    });

    it('changes date range to this month on click', () => {
      const selectMonth = jest.fn();
      const component = renderComponent({ selectMonth });
      component.find(Button).at(1).simulate('click');
      expect(selectMonth).toBeCalled();
    });

    it('changes date range to the last three months on click', () => {
      const selectThreeMonth = jest.fn();
      const component = renderComponent({ selectThreeMonth });
      component.find(Button).at(2).simulate('click');
      expect(selectThreeMonth).toBeCalled();
    });

    it('changes date range to the last year on click', () => {
      const selectYear = jest.fn();
      const component = renderComponent({ selectYear });
      component.find(Button).at(3).simulate('click');
      expect(selectYear).toBeCalled();
    });

    it('changes date range to "all time" on click', () => {
      const selectAll = jest.fn();
      const component = renderComponent({ selectAll });
      component.find(Button).at(4).simulate('click');
      expect(selectAll).toBeCalled();
    });
  });
});