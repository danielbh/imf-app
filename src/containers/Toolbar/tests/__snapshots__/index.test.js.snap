// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Toolbar /> has expected structure and styles 1`] = `
.c0 .toolbar {
  display: -webkit-flex;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-justify-content: flex-end;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
}

<div
  className="c0"
>
  <div
    className="toolbar btn-toolbar"
    role="toolbar"
  >
    <div
      className="btn-group"
    >
      <button
        className="btn btn-default"
        disabled={false}
        onClick={undefined}
        type="button"
      >
        1W
      </button>
      <button
        className="btn btn-default"
        disabled={false}
        onClick={undefined}
        type="button"
      >
        1M
      </button>
      <button
        className="btn btn-default"
        disabled={false}
        onClick={undefined}
        type="button"
      >
        3M
      </button>
      <button
        className="btn btn-default"
        disabled={false}
        onClick={undefined}
        type="button"
      >
        1Y
      </button>
      <button
        className="btn btn-default"
        disabled={false}
        onClick={undefined}
        type="button"
      >
        All
      </button>
    </div>
  </div>
</div>
`;
